domain/consensus/processes/dagtraversalmanager/window.go

```diff
@@ -10,9 +10,37 @@ import (
 // blocks in the past of highHash, the sorting is unspecified.
 // If the number of blocks in the past of startingNode is less then windowSize,
 func (dtm *dagTraversalManager) BlockWindow(stagingArea *model.StagingArea, highHash *externalapi.DomainHash, windowSize int) ([]*externalapi.DomainHash, error) {
+	windowHeap, err := dtm.blockWindowHeap(stagingArea, highHash, windowSize)
+	if err != nil {
+		return nil, err
+	}
+
+	window := make([]*externalapi.DomainHash, 0, len(windowHeap.impl.slice))
+	for _, b := range windowHeap.impl.slice {
+		window = append(window, b.Hash)
+	}
+	return window, nil
+}
+
+func (dtm *dagTraversalManager) BlockWindowWithGHOSTDAGData(stagingArea *model.StagingArea, highHash *externalapi.DomainHash, windowSize int) ([]*externalapi.BlockGHOSTDAGDataHashPair, error) {
+
+	windowHeap, err := dtm.blockWindowHeap(stagingArea, highHash, windowSize)
+	if err != nil {
+		return nil, err
+	}
 
+	window := make([]*externalapi.BlockGHOSTDAGDataHashPair, 0, len(windowHeap.impl.slice))
+	for _, b := range windowHeap.impl.slice {
+		window = append(window, (*externalapi.BlockGHOSTDAGDataHashPair)(b))
+	}
+	return window, nil
+}
+
+func (dtm *dagTraversalManager) blockWindowHeap(stagingArea *model.StagingArea, highHash *externalapi.DomainHash, windowSize int) (*sizedUpBlockHeap, error) {
```

Comment:
I'd rather if this was called `calculateBlockWindowHeap` or something.

Originally when reading this code I thought this just gives you an empty heap for calculating blockWindow